chromadb/config.py

```diff
@@ -1,35 +1,79 @@
 from pydantic import BaseSettings
-from typing import List
+from typing import Optional, List
+import importlib
+import logging
 
 TELEMETRY_WHITELISTED_SETTINGS = [
     "chroma_db_impl",
     "chroma_api_impl",
     "chroma_server_ssl_enabled",
 ]
 
+logger = logging.getLogger(__name__)
+
+_legacy_config_values = {
+    "duckdb": "chromadb.db.duckdb.DuckDB",
+    "duckdb+parquet": "chromadb.db.duckdb.PersistentDuckDB",
+    "clickhouse": "chromadb.db.clickhouse.Clickhouse",
+    "rest": "chromadb.api.fastapi.FastAPI",
+    "local": "chromadb.api.local.LocalAPI",
+}
+
+
+_impls = {}
```

Comment:
Should the impls be shared across the Settings? Unless I'm mistaken - this makes the instantiated deps singletons which feels odd?